hypertrace-core-graphql-gateway-service-utils/src/main/java/org/hypertrace/core/graphql/utils/gateway/AttributeMapConverter.java

```diff
@@ -27,6 +27,7 @@
   public Single<Map<String, Object>> convert(
       Collection<AttributeRequest> attributes, Map<String, Value> response) {
     return Observable.fromIterable(attributes)
+        .filter(attribute -> !Value.getDefaultInstance().equals(response.get(attribute.alias())))
```

Comment:
Hmm - I'd really love to keep the responses in the map as null (for the same reasons I mentioned, distinguishing between a missing response and a null one).

A few different options that could satisfy that would be:
- Catch the conversation error after line 40
- Change value converter to return a `Single<Optional<Object>>` (ideally it would return a `Maybe`, but that breaks our converter interface)
- Because we're in the utils package, take advantage of package protected and inject the real deal `UnwrappedValueConverter`, add another method to it that returns a `Maybe`, returning an empty maybe in the unsupported case. Then change the existing convert method to wrap that and throw in the empty case so it maintains the same behavior as before.


You can probably tell which one I like based on the length of description :)

Reply:
discussed offline, due to issues in java's map collection (https://bugs.openjdk.java.net/browse/JDK-8148463 ) it's actually a pain to collect null values, and collecting optionals would have wider ramifications. Rather than writing our own map collector implementation and making sure it accounts for other behavior (e.g. duplicate handling, null keys, etc.) we'll just go with this until/unless we have issues.